climada/hazard/trop_cyclone.py

```diff
@@ -216,16 +232,20 @@ def set_from_tracks(self, tracks, centroids=None, description='',
                     LOGGER.info("Progress: %d%%", perc)
                     last_perc = perc
                 tc_haz.append(
-                    self._tc_from_track(track, centroids, coastal_idx,
-                                        model=model, store_windfields=store_windfields,
-                                        metric=metric))
+                    TropCyclone.from_single_track(track, centroids, coastal_idx,
+                                                  model=model, store_windfields=store_windfields,
+                                                  metric=metric, intensity_thres=intensity_thres))
             if last_perc < 100:
                 LOGGER.info("Progress: 100%")
-        LOGGER.debug('Append events.')
-        self.__dict__ = TropCyclone.concat([self] + tc_haz).__dict__
+
+        LOGGER.debug('Concatenate events.')
+        haz = TropCyclone.concat(tc_haz)
+        haz.pool = pool
```

Comment:
Is this still needed? What is the use of having the pool attached to self with the static method approach?

Reply:
Currently, the pool attribute is a standard attribute of the `Hazard` base class. We should retain it for consistency with the base class and for backwards compatibility. Users might have code where the pool that has been used to generate a Hazard is reused later and accessed through the `pool` attribute.